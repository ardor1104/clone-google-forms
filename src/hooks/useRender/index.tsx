import { BrowserRouter } from 'react-router-dom';

import renderWithProviders from 'utils/renderWithProviders';

import { ChildrenPropType } from 'components/globalType';
import ThemeTemplate from 'components/templates/ThemeTemplate';

const useRender = (children: ChildrenPropType) => {
  return renderWithProviders(
    <BrowserRouter>
      <ThemeTemplate>{children}</ThemeTemplate>
    </BrowserRouter>,
  );
};

export default useRender;
