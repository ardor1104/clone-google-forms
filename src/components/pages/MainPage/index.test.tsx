import { composeStories } from '@storybook/testing-react';

import useRender from 'hooks/useRender';

import * as stories from './index.stories';

const { Default } = composeStories(stories);

describe('pages/MainPage', () => {
  it('render Default', async () => {
    const component = useRender(<Default />);

    expect(component).toMatchSnapshot();
  });
});
