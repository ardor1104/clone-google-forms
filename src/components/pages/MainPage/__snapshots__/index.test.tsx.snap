// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/MainPage render Default 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jvSYwb egsQDn"
      >
        <div
          class="sc-gzrROc sc-gkSfol eUlCEp eBBxNq"
        >
          <div
            class="sc-fmPOXC jtJwXo"
          >
            <p
              class="sc-eYqcxL cAJNHa"
            >
              최근 설문지
            </p>
            <div
              class="sc-iFoMEM jbmWuP"
            >
              <div
                class="sc-AHaJN cEKjJY"
              >
                <div
                  class="sc-khsqcC dXRRGA"
                >
                  <div
                    class="sc-eVspGN LhmYi"
                  >
                    <p>
                      모든 항목
                    </p>
                    <span
                      class="sc-hLBbgP gNfMnh"
                      width="24"
                    >
                      <svg
                        fill="none"
                        focusable="false"
                      />
                    </span>
                  </div>
                </div>
                <div
                  class="sc-jfTVlA hDAToT"
                />
              </div>
            </div>
            <div
              class="sc-kSGOQU fzVPyi"
            >
              <button
                class="sc-dkrFOg jOmIxm sc-eDvSVe bIQvjX"
                kind="none"
                role="button"
              >
                <span
                  class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                  width="24"
                >
                  <svg
                    fill="none"
                    focusable="false"
                  />
                </span>
              </button>
              <div
                class="sc-AHaJN cEKjJY"
              >
                <div
                  class="sc-khsqcC dXRRGA"
                >
                  <button
                    class="sc-dkrFOg jOmIxm sc-eDvSVe bIQvjX"
                    kind="none"
                    role="button"
                  >
                    <span
                      class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                      width="24"
                    >
                      <svg
                        fill="none"
                        focusable="false"
                      />
                    </span>
                  </button>
                </div>
                <div
                  class="sc-jfTVlA hDAToT"
                />
              </div>
              <button
                class="sc-dkrFOg eKmaHV sc-eDvSVe bIQvjX"
                kind="none"
                role="button"
              >
                <span
                  class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                  width="24"
                >
                  <svg
                    fill="none"
                    focusable="false"
                  />
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-gzrROc eUlCEp"
        />
        <div
          class="sc-edLOhm eOLquQ"
        >
          <div
            class="sc-dkcEsn ktXvPW"
          >
            <a
              class="sc-jKvnYE fEuSky"
            >
              <div
                class="sc-bTUVah dantqo"
              >
                <img
                  alt="FormThumbnail"
                  src="https://thumbs.dreamstime.com/b/flowers-composition-pattern-made-chamomiles-petals-pastel-purple-background-spring-summer-concept-flat-lay-top-view-copy-144695810.jpg"
                />
              </div>
              <div
                class="sc-uokDu kMYdzx"
              >
                <h6
                  class="sc-dTjBdT zeiAL"
                >
                  제목 없는 설문지4
                </h6>
                <div
                  class="sc-hZNxer fXPxGo"
                >
                  <span
                    class="sc-hLBbgP gNfMnh sc-cHGmPC cWedVa"
                    width="24"
                  >
                    <svg
                      fill="none"
                      focusable="false"
                    />
                  </span>
                  <p
                    class="sc-JHWBx eFrWRc"
                  >
                    2023. 5. 6.
                  </p>
                </div>
                <div
                  class="sc-gGWvLE DlqHp"
                >
                  <div
                    class="sc-AHaJN cEKjJY"
                  >
                    <div
                      class="sc-khsqcC dXRRGA"
                    >
                      <button
                        class="sc-dkrFOg jOmIxm sc-eDvSVe deBBD"
                        kind="none"
                        role="button"
                      >
                        <span
                          class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                          width="24"
                        >
                          <svg
                            fill="none"
                            focusable="false"
                          />
                        </span>
                      </button>
                    </div>
                    <div
                      class="sc-jfTVlA hDAToT"
                    />
                  </div>
                </div>
              </div>
            </a>
            <a
              class="sc-jKvnYE fEuSky"
            >
              <div
                class="sc-bTUVah dantqo"
              >
                <img
                  alt="FormThumbnail"
                  src="https://wallpaper.dog/large/20538421.jpg"
                />
              </div>
              <div
                class="sc-uokDu kMYdzx"
              >
                <h6
                  class="sc-dTjBdT zeiAL"
                >
                  제목 없는 설문지5
                </h6>
                <div
                  class="sc-hZNxer fXPxGo"
                >
                  <span
                    class="sc-hLBbgP gNfMnh sc-cHGmPC cWedVa"
                    width="24"
                  >
                    <svg
                      fill="none"
                      focusable="false"
                    />
                  </span>
                  <p
                    class="sc-JHWBx eFrWRc"
                  >
                    2023. 5. 3.
                  </p>
                </div>
                <div
                  class="sc-gGWvLE DlqHp"
                >
                  <div
                    class="sc-AHaJN cEKjJY"
                  >
                    <div
                      class="sc-khsqcC dXRRGA"
                    >
                      <button
                        class="sc-dkrFOg jOmIxm sc-eDvSVe deBBD"
                        kind="none"
                        role="button"
                      >
                        <span
                          class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                          width="24"
                        >
                          <svg
                            fill="none"
                            focusable="false"
                          />
                        </span>
                      </button>
                    </div>
                    <div
                      class="sc-jfTVlA hDAToT"
                    />
                  </div>
                </div>
              </div>
            </a>
            <a
              class="sc-jKvnYE fEuSky"
            >
              <div
                class="sc-bTUVah dantqo"
              >
                <img
                  alt="FormThumbnail"
                  src="https://papers.co/wallpaper/papers.co-so75-blur-gradation-pink-purple-pastel-6-wallpaper.jpg"
                />
              </div>
              <div
                class="sc-uokDu kMYdzx"
              >
                <h6
                  class="sc-dTjBdT zeiAL"
                >
                  제목 없는 설문지3
                </h6>
                <div
                  class="sc-hZNxer fXPxGo"
                >
                  <span
                    class="sc-hLBbgP gNfMnh sc-cHGmPC cWedVa"
                    width="24"
                  >
                    <svg
                      fill="none"
                      focusable="false"
                    />
                  </span>
                  <p
                    class="sc-JHWBx eFrWRc"
                  >
                    2023. 4. 3.
                  </p>
                </div>
                <div
                  class="sc-gGWvLE DlqHp"
                >
                  <div
                    class="sc-AHaJN cEKjJY"
                  >
                    <div
                      class="sc-khsqcC dXRRGA"
                    >
                      <button
                        class="sc-dkrFOg jOmIxm sc-eDvSVe deBBD"
                        kind="none"
                        role="button"
                      >
                        <span
                          class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                          width="24"
                        >
                          <svg
                            fill="none"
                            focusable="false"
                          />
                        </span>
                      </button>
                    </div>
                    <div
                      class="sc-jfTVlA hDAToT"
                    />
                  </div>
                </div>
              </div>
            </a>
            <a
              class="sc-jKvnYE fEuSky"
            >
              <div
                class="sc-bTUVah dantqo"
              >
                <img
                  alt="FormThumbnail"
                  src="https://images.unsplash.com/photo-1629196914168-3a2652305f9f?ixlib=rb-4.0.3"
                />
              </div>
              <div
                class="sc-uokDu kMYdzx"
              >
                <h6
                  class="sc-dTjBdT zeiAL"
                >
                  제목 없는 설문지2
                </h6>
                <div
                  class="sc-hZNxer fXPxGo"
                >
                  <span
                    class="sc-hLBbgP gNfMnh sc-cHGmPC cWedVa"
                    width="24"
                  >
                    <svg
                      fill="none"
                      focusable="false"
                    />
                  </span>
                  <p
                    class="sc-JHWBx eFrWRc"
                  >
                    2023. 4. 2.
                  </p>
                </div>
                <div
                  class="sc-gGWvLE DlqHp"
                >
                  <div
                    class="sc-AHaJN cEKjJY"
                  >
                    <div
                      class="sc-khsqcC dXRRGA"
                    >
                      <button
                        class="sc-dkrFOg jOmIxm sc-eDvSVe deBBD"
                        kind="none"
                        role="button"
                      >
                        <span
                          class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                          width="24"
                        >
                          <svg
                            fill="none"
                            focusable="false"
                          />
                        </span>
                      </button>
                    </div>
                    <div
                      class="sc-jfTVlA hDAToT"
                    />
                  </div>
                </div>
              </div>
            </a>
            <a
              class="sc-jKvnYE fEuSky"
            >
              <div
                class="sc-bTUVah dantqo"
              >
                <img
                  alt="FormThumbnail"
                  src="https://i.pinimg.com/originals/d0/b7/2c/d0b72c0f7022546b01e13d545f0e66fc.jpg"
                />
              </div>
              <div
                class="sc-uokDu kMYdzx"
              >
                <h6
                  class="sc-dTjBdT zeiAL"
                >
                  제목 없는 설문지1
                </h6>
                <div
                  class="sc-hZNxer fXPxGo"
                >
                  <span
                    class="sc-hLBbgP gNfMnh sc-cHGmPC cWedVa"
                    width="24"
                  >
                    <svg
                      fill="none"
                      focusable="false"
                    />
                  </span>
                  <p
                    class="sc-JHWBx eFrWRc"
                  >
                    2023. 4. 1.
                  </p>
                </div>
                <div
                  class="sc-gGWvLE DlqHp"
                >
                  <div
                    class="sc-AHaJN cEKjJY"
                  >
                    <div
                      class="sc-khsqcC dXRRGA"
                    >
                      <button
                        class="sc-dkrFOg jOmIxm sc-eDvSVe deBBD"
                        kind="none"
                        role="button"
                      >
                        <span
                          class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                          width="24"
                        >
                          <svg
                            fill="none"
                            focusable="false"
                          />
                        </span>
                      </button>
                    </div>
                    <div
                      class="sc-jfTVlA hDAToT"
                    />
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jvSYwb egsQDn"
    >
      <div
        class="sc-gzrROc sc-gkSfol eUlCEp eBBxNq"
      >
        <div
          class="sc-fmPOXC jtJwXo"
        >
          <p
            class="sc-eYqcxL cAJNHa"
          >
            최근 설문지
          </p>
          <div
            class="sc-iFoMEM jbmWuP"
          >
            <div
              class="sc-AHaJN cEKjJY"
            >
              <div
                class="sc-khsqcC dXRRGA"
              >
                <div
                  class="sc-eVspGN LhmYi"
                >
                  <p>
                    모든 항목
                  </p>
                  <span
                    class="sc-hLBbgP gNfMnh"
                    width="24"
                  >
                    <svg
                      fill="none"
                      focusable="false"
                    />
                  </span>
                </div>
              </div>
              <div
                class="sc-jfTVlA hDAToT"
              />
            </div>
          </div>
          <div
            class="sc-kSGOQU fzVPyi"
          >
            <button
              class="sc-dkrFOg jOmIxm sc-eDvSVe bIQvjX"
              kind="none"
              role="button"
            >
              <span
                class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                width="24"
              >
                <svg
                  fill="none"
                  focusable="false"
                />
              </span>
            </button>
            <div
              class="sc-AHaJN cEKjJY"
            >
              <div
                class="sc-khsqcC dXRRGA"
              >
                <button
                  class="sc-dkrFOg jOmIxm sc-eDvSVe bIQvjX"
                  kind="none"
                  role="button"
                >
                  <span
                    class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                    width="24"
                  >
                    <svg
                      fill="none"
                      focusable="false"
                    />
                  </span>
                </button>
              </div>
              <div
                class="sc-jfTVlA hDAToT"
              />
            </div>
            <button
              class="sc-dkrFOg eKmaHV sc-eDvSVe bIQvjX"
              kind="none"
              role="button"
            >
              <span
                class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                width="24"
              >
                <svg
                  fill="none"
                  focusable="false"
                />
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-gzrROc eUlCEp"
      />
      <div
        class="sc-edLOhm eOLquQ"
      >
        <div
          class="sc-dkcEsn ktXvPW"
        >
          <a
            class="sc-jKvnYE fEuSky"
          >
            <div
              class="sc-bTUVah dantqo"
            >
              <img
                alt="FormThumbnail"
                src="https://thumbs.dreamstime.com/b/flowers-composition-pattern-made-chamomiles-petals-pastel-purple-background-spring-summer-concept-flat-lay-top-view-copy-144695810.jpg"
              />
            </div>
            <div
              class="sc-uokDu kMYdzx"
            >
              <h6
                class="sc-dTjBdT zeiAL"
              >
                제목 없는 설문지4
              </h6>
              <div
                class="sc-hZNxer fXPxGo"
              >
                <span
                  class="sc-hLBbgP gNfMnh sc-cHGmPC cWedVa"
                  width="24"
                >
                  <svg
                    fill="none"
                    focusable="false"
                  />
                </span>
                <p
                  class="sc-JHWBx eFrWRc"
                >
                  2023. 5. 6.
                </p>
              </div>
              <div
                class="sc-gGWvLE DlqHp"
              >
                <div
                  class="sc-AHaJN cEKjJY"
                >
                  <div
                    class="sc-khsqcC dXRRGA"
                  >
                    <button
                      class="sc-dkrFOg jOmIxm sc-eDvSVe deBBD"
                      kind="none"
                      role="button"
                    >
                      <span
                        class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                        width="24"
                      >
                        <svg
                          fill="none"
                          focusable="false"
                        />
                      </span>
                    </button>
                  </div>
                  <div
                    class="sc-jfTVlA hDAToT"
                  />
                </div>
              </div>
            </div>
          </a>
          <a
            class="sc-jKvnYE fEuSky"
          >
            <div
              class="sc-bTUVah dantqo"
            >
              <img
                alt="FormThumbnail"
                src="https://wallpaper.dog/large/20538421.jpg"
              />
            </div>
            <div
              class="sc-uokDu kMYdzx"
            >
              <h6
                class="sc-dTjBdT zeiAL"
              >
                제목 없는 설문지5
              </h6>
              <div
                class="sc-hZNxer fXPxGo"
              >
                <span
                  class="sc-hLBbgP gNfMnh sc-cHGmPC cWedVa"
                  width="24"
                >
                  <svg
                    fill="none"
                    focusable="false"
                  />
                </span>
                <p
                  class="sc-JHWBx eFrWRc"
                >
                  2023. 5. 3.
                </p>
              </div>
              <div
                class="sc-gGWvLE DlqHp"
              >
                <div
                  class="sc-AHaJN cEKjJY"
                >
                  <div
                    class="sc-khsqcC dXRRGA"
                  >
                    <button
                      class="sc-dkrFOg jOmIxm sc-eDvSVe deBBD"
                      kind="none"
                      role="button"
                    >
                      <span
                        class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                        width="24"
                      >
                        <svg
                          fill="none"
                          focusable="false"
                        />
                      </span>
                    </button>
                  </div>
                  <div
                    class="sc-jfTVlA hDAToT"
                  />
                </div>
              </div>
            </div>
          </a>
          <a
            class="sc-jKvnYE fEuSky"
          >
            <div
              class="sc-bTUVah dantqo"
            >
              <img
                alt="FormThumbnail"
                src="https://papers.co/wallpaper/papers.co-so75-blur-gradation-pink-purple-pastel-6-wallpaper.jpg"
              />
            </div>
            <div
              class="sc-uokDu kMYdzx"
            >
              <h6
                class="sc-dTjBdT zeiAL"
              >
                제목 없는 설문지3
              </h6>
              <div
                class="sc-hZNxer fXPxGo"
              >
                <span
                  class="sc-hLBbgP gNfMnh sc-cHGmPC cWedVa"
                  width="24"
                >
                  <svg
                    fill="none"
                    focusable="false"
                  />
                </span>
                <p
                  class="sc-JHWBx eFrWRc"
                >
                  2023. 4. 3.
                </p>
              </div>
              <div
                class="sc-gGWvLE DlqHp"
              >
                <div
                  class="sc-AHaJN cEKjJY"
                >
                  <div
                    class="sc-khsqcC dXRRGA"
                  >
                    <button
                      class="sc-dkrFOg jOmIxm sc-eDvSVe deBBD"
                      kind="none"
                      role="button"
                    >
                      <span
                        class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                        width="24"
                      >
                        <svg
                          fill="none"
                          focusable="false"
                        />
                      </span>
                    </button>
                  </div>
                  <div
                    class="sc-jfTVlA hDAToT"
                  />
                </div>
              </div>
            </div>
          </a>
          <a
            class="sc-jKvnYE fEuSky"
          >
            <div
              class="sc-bTUVah dantqo"
            >
              <img
                alt="FormThumbnail"
                src="https://images.unsplash.com/photo-1629196914168-3a2652305f9f?ixlib=rb-4.0.3"
              />
            </div>
            <div
              class="sc-uokDu kMYdzx"
            >
              <h6
                class="sc-dTjBdT zeiAL"
              >
                제목 없는 설문지2
              </h6>
              <div
                class="sc-hZNxer fXPxGo"
              >
                <span
                  class="sc-hLBbgP gNfMnh sc-cHGmPC cWedVa"
                  width="24"
                >
                  <svg
                    fill="none"
                    focusable="false"
                  />
                </span>
                <p
                  class="sc-JHWBx eFrWRc"
                >
                  2023. 4. 2.
                </p>
              </div>
              <div
                class="sc-gGWvLE DlqHp"
              >
                <div
                  class="sc-AHaJN cEKjJY"
                >
                  <div
                    class="sc-khsqcC dXRRGA"
                  >
                    <button
                      class="sc-dkrFOg jOmIxm sc-eDvSVe deBBD"
                      kind="none"
                      role="button"
                    >
                      <span
                        class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                        width="24"
                      >
                        <svg
                          fill="none"
                          focusable="false"
                        />
                      </span>
                    </button>
                  </div>
                  <div
                    class="sc-jfTVlA hDAToT"
                  />
                </div>
              </div>
            </div>
          </a>
          <a
            class="sc-jKvnYE fEuSky"
          >
            <div
              class="sc-bTUVah dantqo"
            >
              <img
                alt="FormThumbnail"
                src="https://i.pinimg.com/originals/d0/b7/2c/d0b72c0f7022546b01e13d545f0e66fc.jpg"
              />
            </div>
            <div
              class="sc-uokDu kMYdzx"
            >
              <h6
                class="sc-dTjBdT zeiAL"
              >
                제목 없는 설문지1
              </h6>
              <div
                class="sc-hZNxer fXPxGo"
              >
                <span
                  class="sc-hLBbgP gNfMnh sc-cHGmPC cWedVa"
                  width="24"
                >
                  <svg
                    fill="none"
                    focusable="false"
                  />
                </span>
                <p
                  class="sc-JHWBx eFrWRc"
                >
                  2023. 4. 1.
                </p>
              </div>
              <div
                class="sc-gGWvLE DlqHp"
              >
                <div
                  class="sc-AHaJN cEKjJY"
                >
                  <div
                    class="sc-khsqcC dXRRGA"
                  >
                    <button
                      class="sc-dkrFOg jOmIxm sc-eDvSVe deBBD"
                      kind="none"
                      role="button"
                    >
                      <span
                        class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                        width="24"
                      >
                        <svg
                          fill="none"
                          focusable="false"
                        />
                      </span>
                    </button>
                  </div>
                  <div
                    class="sc-jfTVlA hDAToT"
                  />
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
